src/Controller/GameController.php

```diff
@@ -3,26 +3,135 @@
 
 use App\Entity\Game;
 use App\Entity\Ship;
-use App\Constants\GameConstants;
+use App\Entity\Shoot;
 use App\GameManipulator;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
+use App\Form\Type\ShootShipType;
+use App\Form\Type\AbandonGameType;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Contracts\Translation\TranslatorInterface;
 
 class GameController extends AbstractController
 {
     private $gameManipulator;
 
-    public function __construct(GameManipulator $gameManipulator)
+    private $translator;
+
+    public function __construct(GameManipulator $gameManipulator, TranslatorInterface $translator)
     {
         $this->gameManipulator = $gameManipulator;
+        $this->translator = $translator;
     }
 
-    public function random(): Response
-    {        
-        $game = $this->gameManipulator->createGame();
+    /**
+     * @ParamConverter("game", options={"mapping": {"hash": "hash"}})
+     */
+    public function index(Request $request, Game $game): Response
+    {
+        if ($game->isAbandoned()) {
+            return $this->render('abandoned.html.twig');
+        }
+
+        $abandonForm = $this->createForm(AbandonGameType::class);
+        $abandonForm->handleRequest($request);
+
+        if ($abandonForm->isSubmitted() && $abandonForm->isValid()) {
+            $this->gameManipulator->abandonGame($game);
+
+            return $this->redirectToRoute('index');
+        }
+
+        $triggerViews = $this->createTriggersForms($game);
+
+        $trigger = $this->createForm(ShootShipType::class)->handleRequest($request);
+
+        if ($trigger->isSubmitted() && $trigger->isValid()) {
+
```

Comment:
There's to many empty line everywhere. Please group things to make it easier to read